server/service/src/sensor/berlinger.rs

```diff
@@ -0,0 +1,380 @@
+use chrono::NaiveDateTime;
+use repository::{DatetimeFilter, EqualFilter};
+use repository::{
+    RepositoryError, Sensor, SensorFilter, SensorRepository, SensorRow, SensorRowRepository,
+    SensorType, StorageConnection, TemperatureBreach, TemperatureBreachConfig,
+    TemperatureBreachConfigFilter, TemperatureBreachConfigRepository, TemperatureBreachConfigRow,
+    TemperatureBreachConfigRowRepository, TemperatureBreachFilter, TemperatureBreachRepository,
+    TemperatureBreachRow, TemperatureBreachRowRepository, TemperatureBreachRowType, TemperatureLog,
+    TemperatureLogFilter, TemperatureLogRepository, TemperatureLogRow, TemperatureLogRowRepository,
+};
+use util::uuid::uuid;
+
+use crate::{service_provider::ServiceContext, SingleRecordError};
+
+extern crate temperature_sensor;
+use temperature_sensor::*;
+
+pub fn get_breach_row_type(breach_type: &BreachType) -> TemperatureBreachRowType {
```

Comment:
Although it's a bit of a stylistic preference, but like doing implementaiton on the the type. Below is an example of this (since inbound shipment can only be updated to some status, we restrict it in graphql, and then translate to row status):

https://github.com/openmsupply/open-msupply/blob/cfbf37020a26ff900d9a3d3a305f8008d7ded9e4/server/service/src/invoice/inbound_shipment/update/mod.rs#L152-L160

Reply:
Maybe I've misunderstood but when I tried this:

```rust
impl BreachType {
    pub fn breach_row_type(&self) -> TemperatureBreachRowType {
        match self {
            BreachType::ColdConsecutive => TemperatureBreachRowType::ColdConsecutive,
            BreachType::ColdCumulative => TemperatureBreachRowType::ColdCumulative,
            BreachType::HotConsecutive => TemperatureBreachRowType::HotConsecutive,
            BreachType::HotCumulative => TemperatureBreachRowType::HotCumulative,
        }
    }
}
```
it said that I couldn't define `impl` for a type outside the crate, which makes sense as `BreachType` is part of the external `temperature-sensor` crate

Reply:
You a right, they both external types, sorry misleading.